### DISCUSSION TITLE
Avi's Contribution Journal

### DISCUSSION THREAD

**Discussion by avidrucker** [athensresearch/athens#1106] [General]
Hey all. I'd like to share with you my first attempt at contributing to open source. I'm hoping that someone else finds this of value.

Original issue for contribution: https://github.com/athensresearch/athens/issues/649
First pull request: https://github.com/athensresearch/athens/pull/1064

## Current Status:
- [x] Add help icon to top menu bar
- [x] Create temporary popup on click of help icon
- [ ] Remove temporary popup
- [ ] Create modal on click of help icon
- [ ] Display in modal a component with text
- [ ] Render a close button in the upper right corner of the modal
- [ ] Display multiple relevant sections of content in the modal
- [ ] Implement inter-modal navigation (see most recent screenshot above) by @jsmorabito

**Comment by avidrucker** [comment]
## Next Steps
- [ ] Get my PR to pass the tests
   - [x] Learn what I need to do to pass the tests
       - [ ] Resolve linting warning: `src/cljs/athens/views/app_toolbar.cljs:18:6: warning: Unsorted namespace: @material-ui/icons/HelpOutline`
       - [ ] Confirm no linting issues remain by running `script/lint`
       - [x] Find command to prettify code to pass style checks
           - [ ] Run command `cljstyle fix`
           - [ ] Repush & reopen PR to request assistance for further work in pair/group programming 
- [ ] See how other modals inside Athens work 
    - [ ] See modal example: Filesystem dialogue, search `filesystem`

**Reply by avidrucker** [reply]
Quoting from: https://athensresearch.gitbook.io/handbook/development/development
```
To see the problems reported by clj-kondo, run script/lint. Example run:
$ script/lint
linting took 257ms, errors: 0, warnings: 0
Your editor may also be able to integrate with clj-kondo’s output. For example, if you use Calva for VS Code, then clj-kondo’s messages are reported in the Problems panel.
Clojure Styling
To format your code or check that your code is formatted correctly, you will need to use cljstyle. Instructions for installing it are in cljstyle’s README.
To check if your Clojure code is formatted correctly, run cljstyle check. If there is no output and the return code is zero, you’re good. You can also run script/style, but currently it only works if you’re running Linux.
To reformat all your Clojure files in place, run cljstyle fix.
```

**Comment by avidrucker** [comment]
## Questions

**Reply by avidrucker** [reply]
How does one render a modal in reagent?

**Comment by jsmorabito** [comment]
this is a great idea Avi!! one suggestion is that you might like to merge your next steps and questions comments into the main post to keep it all in one nice, little package. unless maybe im misunderstanding your intent?

**Reply by avidrucker** [reply]
I'm experimenting with separate threads, because there may be questions or comments on any given item, and (as much as possible) I want to avoid "mutating" pre-existing items 😉
